services/user.js

```diff
@@ -793,6 +793,17 @@ class UserService {
             updatedUser.notifications = userNotifications;
         }
     }
+
+    async getUsersByStudyID(studyID) {
+        const query = {
+            "studies": {"$in": [studyID, "All"]}
+        }; // user's studies contains studyID
+        const users = await this.userCollection.aggregate([{"$match": query}]);
+        for (const user of users) {
+            user.studies = await this.#findApprovedStudies(user.studies);
```

Comment:
Can we collect all approved IDs from the users and make a single query using **await this.#findApprovedStudies**? Querying for all users could lead to performance issues.

Reply:
Since each user has different studies, need retrieve studies one user by one user.

Reply:
We can start by collecting all the study IDs from the requested users. Then, perform a single query using those study IDs. The results will include these study IDs, which can then be iterated over in the loop for further processing. Is that possible? I think this might solve the performance issue.

Reply:
It's possible but make the code very complex.  I'll use a special query rather than getUserByStudy